### PR TITLE
Added range slider

### PR DESCRIPTION
**Pull request by JeCFe**
Added a generic range sliders component

### CODE REVIEW

File: src/range-slider/RangeSlider.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    },
+    defaultVariants: {
+      size: "medium",
+    },
+  }
+);
+
+type Props = {
+  onVolumeChange: (volume: number) => void;
+  min?: number;
```
Comment (JeCFe) [inline src/range-slider/RangeSlider.tsx:26]: These shouldn't be necessary as they should be provided via the InputHTMLAttributes

File: src/range-slider/RangeSlider.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+} & Omit<InputHTMLAttributes<HTMLInputElement>, "size"> &
+  VariantProps<typeof rangeSlider>;
+
+export const RangeSlider = forwardRef<HTMLInputElement, Props>(
+  (
+    {
+      onVolumeChange,
+      className,
+      size,
+      min = 0,
```
Comment (JeCFe) [inline src/range-slider/RangeSlider.tsx:38]: These are already the default values from the input element

File: src/range-slider/RangeSlider.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+      max = 100,
+      label = undefined,
+      ...rest
+    },
+    ref
+  ) => {
+    const handleVolumeChange = (event: ChangeEvent<HTMLInputElement>) => {
```
Comment (JeCFe) [inline src/range-slider/RangeSlider.tsx:45]: I'm in a debate about whether there is any value in this forward ref. 
The reason for the inclusion is the case of using a state management lib such as `react-hook-form` could allow the state management lib to control the range state via a ref. 

I'm not precious about keeping it, might be removed in the future is if this is providing no value 



Tl;Dr might need forward ref in future, API is easy to use from the beginning than add onto an existing component, but is very easy to component imo
